### PR TITLE
Add more CRD validation

### PR DESCRIPTION
**Pull request by thockin**
It's not all done but the fields we are playing with now are mostly done.

### CODE REVIEW

File: pkg/api/v1alpha1/actortemplate_types.go [hunk 1]
```diff
@@ -55,18 +64,29 @@ type Container struct {
 // literal values are not expanded with Kubernetes-style $(VAR) references,
 // envFrom is not supported, and valueFrom currently supports only secretKeyRef.
 //
-// +kubebuilder:validation:XValidation:rule="!(has(self.value) && has(self.valueFrom))",message="value and valueFrom are mutually exclusive"
+// +kubebuilder:validation:ExactlyOneOf={value, valueFrom}
```
Comment (juli4n) [inline pkg/api/v1alpha1/actortemplate_types.go:67]: How does this interplay with `omitempty` in `Value`?  Can you add a test where we set `Env` to an empty string?

Comment (thockin) [reply pkg/api/v1alpha1/actortemplate_types.go:67]: You are correct!

Kubectl works because it never serializes to a Go type, which is than builtin types here.  In CRD `optional` is satisfied by the key being present at all, whereas builtins require an actual value (since builtin validation happens on the deserialized data).

This means we have to change Value to a pointer.  Will add that here.

**Comment by thockin** [comment]
@EItanya specifically for the `Value` change.  I missed that in review!

**Comment by EItanya** [comment]
> Eitan Yarmush (@EItanya) specifically for the `Value` change. I missed that in review!

Good catch, should we think about using the linter: https://github.com/kubernetes-sigs/kube-api-linter

**Review by juli4n — APPROVED** [review]

**Comment by juli4n** [comment]
> > Eitan Yarmush (Eitan Yarmush (@EItanya)) specifically for the `Value` change. I missed that in review!
> 
> Good catch, should we think about using the linter: https://github.com/kubernetes-sigs/kube-api-linter

This came up a few days ago as well in another PR. Filed https://github.com/agent-substrate/substrate/issues/207 to track.
